orttraining/orttraining/training_ops/cuda/communication/common.h

```diff
@@ -12,12 +12,17 @@ typedef struct {
   // It should be removed once we enable CUDA-aware MPI.
   void* buffer;
   // The size of buffer's content in bytes.
-  size_t size;
+  int size;
```

Comment:
This limits to the buffer to be smaller than 2GB. It's fine if throw when overflow happens.

Reply:
MPI spec takes only int. 
We added safety checks.